abi.go

```diff
@@ -44,25 +50,42 @@ func newMapABIFromFd(fd *internal.FD) (string, *MapABI, error) {
 
 	return "", &MapABI{
 		MapType(info.mapType),
+		MapID(info.id),
 		info.keySize,
 		info.valueSize,
 		info.maxEntries,
 		info.flags,
+		internal.CString(info.mapName[:]),
+		UnspecifiedProgram,
 	}, nil
 }
 
 func newMapABIFromProc(fd *internal.FD) (*MapABI, error) {
 	var abi MapABI
-	err := scanFdInfo(fd, map[string]interface{}{
+	fields := map[string]interface{}{
 		"map_type":    &abi.Type,
 		"key_size":    &abi.KeySize,
 		"value_size":  &abi.ValueSize,
 		"max_entries": &abi.MaxEntries,
 		"map_flags":   &abi.Flags,
-	})
+		"map_id":      &abi.ID,
+	}
+	if abi.Type == ProgramArray {
+		fields["owner_prog_type"] = &abi.OwnerProgType
+	}
+
+	err := scanFdInfo(fd, fields)
+	if xerrors.Is(err, errMissingFields) {
+		return nil, &internal.UnsupportedFeatureError{
+			Name:           "reading map ABI from /proc/self/fdinfo",
+			MinimumVersion: internal.Version{4, 5, 0},
```

Comment:
Not all fields are available from 4.5, see the failing test on 4.9.

Reply:
Is there a way to extract fields based on kernel version?

Reply:
It seems the `map_id` field was only added in Kernel 4.18: torvalds/linux@4316b40914ecd